File: round1076-div3/problem-c.py
t = int(input())

for i in range(t):
    n, q = map(int, input().split())
    a = list(map(int, input().split()))
    b = list(map(int, input().split()))

    c = [max(a[i], b[i]) for i in range(n)]

    suf = [0] * n
    suf[-1] = c[-1]
    for i in range(n - 2, -1, -1):
        suf[i] = max(suf[i + 1], c[i])

    for i in range(q):
        l, r = map(int, input().split())
        l -= 1
        r -= 1
        print(sum(suf[l:r + 1]), end=" ")
    print()

File: round1076-div3/problem-c.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int t;
    cin >> t;

    while (t--) {
        int n, q;
        cin >> n >> q;

        vector<long long> a(n), b(n);
        for (int i = 0; i < n; i++) cin >> a[i];
        for (int i = 0; i < n; i++) cin >> b[i];

        vector<long long> c(n);
        for (int i = 0; i < n; i++) {
            c[i] = max(a[i], b[i]);
        }

        vector<long long> suf(n);
        suf[n - 1] = c[n - 1];
        for (int i = n - 2; i >= 0; i--) {
            suf[i] = max(suf[i + 1], c[i]);
        }

        vector<long long> pref(n + 1, 0);
        for (int i = 0; i < n; i++) {
            pref[i + 1] = pref[i] + suf[i];
        }

        while (q--) {
            int l, r;
            cin >> l >> r;
            l--; r--;
            cout << pref[r + 1] - pref[l] << " ";
        }
        cout << "\n";
    }

    return 0;
}
